lib/services/hooks-api-service.js

```diff
@@ -0,0 +1,91 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = hookApiServiceFactory;
+di.annotate(hookApiServiceFactory, new di.Provide('Http.Services.Api.Hooks'));
+di.annotate(hookApiServiceFactory,
+    new di.Inject(
+        'Services.Waterline',
+        'Logger',
+        '_',
+        'Errors'
+    )
+);
+function hookApiServiceFactory(
+    waterline,
+    Logger,
+    _,
+    Errors
+) {
+    var logger = Logger.initialize(hookApiServiceFactory);
+    function HooksApiService() {
+    }
+
+    /**
+     * Get hooks
+     * @return {Promise}
+     */
+    HooksApiService.prototype.getHooks = function (query) {
+        return waterline.hooks.find(query);
+    };
+
+    /**
+     * Create new hooks
+     * @param {Object} [req] HTTP request
+     * @param {Object} [res] HTTP response
+     * @return
+     */
+    HooksApiService.prototype.createHook = function(body) {
+        var self = this;
+        return waterline.hooks.findOne({url: body.url})
+        .then(function(entry){
```

Comment:
What happens if there are multiple hooks with the same name?  Looking at this PR: https://github.com/RackHD/on-core/pull/238/files it doesn't appear that the `name` property is uniquely indexed.